addons/survey/models/survey_survey.py

```diff
@@ -427,11 +427,10 @@ def _check_survey_responsible_access(self):
                 - and there is a survey responsible,
                 - and this responsible is not survey manager (just survey officer),
             check the responsible is part of the list."""
-        surveys_to_check = self.filtered(lambda s: bool(s.user_id - s.restrict_user_ids))
-        if surveys_to_check:
-            valid_surveys = surveys_to_check._filtered_access("write")
-            failing_surveys_sudo = (self - valid_surveys).sudo()
-            if failing_surveys_sudo:
+        for user_id, surveys in self.filtered(lambda s: bool(s.user_id - s.restrict_user_ids)).grouped('user_id').items():
+            accessible = surveys.with_env(surveys.env(user=user_id, su=False))._filtered_access("write")
+            if len(accessible) < len(surveys):
+                failing_surveys_sudo = (self - accessible).sudo()
```

Comment:
self is already in sudo then the extra sudo isn't need IMO

Reply:
Indeed, now it is in sudo, but for readability (and if you were to call the function directly), I would leave it here.